stock-usecase/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/DropdownComponents.kt

```diff
@@ -473,17 +478,20 @@ fun openOrgUnitTreeSelector(
         .showAsDialog()
         .singleSelection()
         .orgUnitScope(OrgUnitSelectorScope.ProgramCaptureScope(settingsUiState.programUid))
-        .withPreselectedOrgUnits(orgUnitData?.let { listOf(it.uid()) } ?: emptyList())
+//        .withPreselectedOrgUnits(orgUnitData?.let { listOf(it.uid()) } ?: emptyList())
+        .withPreselectedOrgUnits(
+            viewModel.orgUnitData.value?.let { listOf(it.uid()) } ?: emptyList()
+        )
         .onSelection { selectedOrgUnits ->
             val selectedOrgUnit = selectedOrgUnits.firstOrNull()
             if (selectedOrgUnit != null) {
-                if (orgUnitData != selectedOrgUnit && hasUnsavedData) {
+                if (viewModel.orgUnitData.value != selectedOrgUnit && hasUnsavedData) {
                     launchDialog.invoke(R.string.transaction_discarted) { result ->
                         when (result) {
                             EditionDialogResult.DISCARD -> {
                                 // Perform the transaction change and clear data
                                 onFacilitySelected.invoke(selectedOrgUnit)
-                                orgUnitData = selectedOrgUnit
```

Comment:
I think that is a good idea to remove it from the view and use it from the viewModel.
Given that you are removing the usage of it, why don't you try to remove the orgUnitData property from this class?. Maybe you can remove also the orgUnitName prperty as weel and get it from the viewmodel by `viewModel.settingUiModel.facilityName`
What do you think?

Reply:
I was waiting for this comment, because I saw that this variable consumes memory of the app, even after the module closes. So I just wanted to make this ticket works and cover the functionality. The idea is good, and we will have to change all implementations of this var.  Good idea Andres.